src/vdaf/prio2.rs

```
@@ -0,0 +1,372 @@
+    const PRIV_KEY2: &str = "BNNOqoU54GPo+1gTPv+hCgA9U2ZCKd76yOMrWa1xTWgeb4LhFLMQIQoRwDVaW64g/WTdcxT4rDULoycUNFB60LER6hPEHg/ObBnRPV1rwS3nj9Bj0tbjVPPyL9p8QW8B+w==";
+
+    #[test]
+    fn run_prio2() {
+        let prio2 = Prio2::new(6).unwrap();
+
+        assert_eq!(
+            run_vdaf(
+                &prio2,
+                &(),
+                [
+                    vec![0, 0, 0, 0, 1, 0],
+                    vec![0, 1, 0, 0, 0, 0],
+                    vec![0, 1, 1, 0, 0, 0],
+                    vec![1, 1, 1, 0, 0, 0],
+                    vec![0, 0, 0, 0, 1, 1],
+                ]
+            )
+            .unwrap(),
+            vec![1, 3, 2, 0, 2, 1],
+        );
+    }
+
+    #[test]
+    fn enpa_client_interop() {
+        let mut rng = thread_rng();
+        let priv_key1 = PrivateKey::from_base64(PRIV_KEY1).unwrap();
+        let priv_key2 = PrivateKey::from_base64(PRIV_KEY2).unwrap();
+        let pub_key1 = PublicKey::from(&priv_key1);
+        let pub_key2 = PublicKey::from(&priv_key2);
+
+        let data: Vec<Prio2Field> = [0, 0, 1, 1, 0]
+            .iter()
+            .map(|x| Prio2Field::from(*x))
+            .collect();
+        let (encrypted_input_share1, encrypted_input_share2) =
+            encode_simple(&data, pub_key1, pub_key2).unwrap();
+
+        let input_share1 = decrypt_share(&encrypted_input_share1, &priv_key1).unwrap();
+        let input_share2 = decrypt_share(&encrypted_input_share2, &priv_key2).unwrap();
+
+        let prio2 = Prio2::new(data.len()).unwrap();
+        let input_shares = vec![
+            Share::get_decoded_with_param(&(&prio2, 0), &input_share1).unwrap(),
+            Share::get_decoded_with_param(&(&prio2, 1), &input_share2).unwrap(),
+        ];
+
+        let verify_key = rng.gen();
+        let mut nonce = [0; 16];
+        rng.fill(&mut nonce);
+        run_vdaf_prepare(&prio2, &verify_key, &(), &nonce, input_shares).expect("test failed");
```

I believe the `.expect` will discard the error from `run_vdaf_prepare`:

```suggestion
        run_vdaf_prepare(&prio2, &verify_key, &(), &nonce, input_shares).unwrap();
```